app/models/unique_name.rb

```diff
@@ -5,17 +5,23 @@
 ###
 
 class UniqueName < ApplicationRecord
-
   def self.update!
     Rails.logger.info 'Updating the unique names table'
 
     # Build double metaphone representations for all names in the database
-    names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.map{ |n| n&.downcase || ''} + GrdaWarehouse::Hud::Client.source.select('LastName').distinct.pluck('LastName')&.map{ |n| n&.downcase || ''}
-    names.uniq.each do |name|
+    existing_names = UniqueName.pluck(:name)
+    all_names = GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')&.
```

Comment:
It was in the code you inherited but I think `GrdaWarehouse::Hud::Client.source.select('FirstName').distinct.pluck('FirstName')` has identical behaviour as `GrdaWarehouse::Hud::Client.source.distinct.pluck('FirstName')` with less repetition

Reply:
Cool. I did wonder about that.